### DISCUSSION TITLE
Using Jq

### DISCUSSION THREAD

**Discussion by globalflea** [serverlessworkflow/specification#494] [Q&A] [answered]
i am new to Jq expression and the command line tool.  Would appreciate help with this.
In the spec, an example is given under the Action Data Filter section with the following:
```
{ "actions":[
    {  "functionRef": "breadAndPastaTypesFunction",
       "actionDataFilter": { 
           "results": "${ [ .breads[0], .pasta[1] ] }",
          "toStateData": "${ .itemsToBuyAtStore }"
       }     }   ] }
```
I was trying to use the jq tool to try evaluating the "results" filter, but somehow can't get the jq tool to parse the expressions, for example:
```
echo '{ "breads": ["baguette", "brioche", "rye"], "pasta": [ "penne",  "spaghetti", "ravioli"] }' | jq '${ [ .breads[0],  .pasta[1] ] }'
```
gives the following error:
```
jq: error: syntax error, unexpected '{', expecting IDENT or __loc__ (Unix shell quoting issues?) at <top-level>, line 1:
${ [ .breads[0],  .pasta[1] ] }
jq: 1 compile error
```
and it seems in the [jq website](https://stedolan.github.io/jq/manual/v1.6/) the example use of "$" is given as below to create a variable which is used subsequently
```
length as $array_length | add / $array_length
```

**Comment by tsurdilo** [comment, marked as answer]
expression syntax in serverless workflow:
${ expression }

this is the syntax regardless of expression lang used.

so try running the actual expression inside

**Reply by globalflea** [reply]
Oh apologies, I was stuck at the v0.6 spec and I thought the spec was giving an example of the jq expression rather than stating that jq expression are embedded within "${}". But isn't this sort of unnecessary, given that we are expecting jq expression already - ie why do we need this extra markers?

**Reply by ricardozanini** [reply]
In some fields, we accept static strings and expressions, so the `${}` helps the runtime implementation to differentiate them. In this case, it is always an expression, but we kept this way to keep the cohesion within the workflow definition.
